src/coreclr/src/jit/gentree.cpp

```diff
@@ -1288,6 +1288,15 @@ bool GenTree::Compare(GenTree* op1, GenTree* op2, bool swapOK)
                     return true;
                 }
                 break;
+
+            case GT_CNS_STR:
+                if ((op1->AsStrCon()->gtSconCPX == op2->AsStrCon()->gtSconCPX) &&
+                    (op1->AsStrCon()->gtScpHnd == op2->AsStrCon()->gtScpHnd))
+                {
+                    return true;
+                }
+                break;
```

Comment:
Is it going to fold comparisons of constant strings? (don't know where `GenTree::Compare` is used)